lib/ansible/modules/source_control/git.py

```diff
@@ -595,14 +595,7 @@ def is_not_a_branch(git_path, module, dest):
     return False
 
 
-def get_head_branch(git_path, module, dest, remote, bare=False):
-    '''
-    Determine what branch HEAD is associated with.  This is partly
-    taken from lib/ansible/utils/__init__.py.  It finds the correct
-    path to .git/HEAD and reads from that file the branch that HEAD is
-    associated with.  In the case of a detached HEAD, this will look
-    up the branch in .git/refs/remotes/<remote>/HEAD.
-    '''
+def get_repo_path(module, dest, bare):
```

Comment:
Returning an error on Ansible module protocol level shouldn't be mixed up with just identifying a directory.
Use exceptions to communicate failures to the calling code and process them there, outside of this function.
So no `module` argument is needed here — let's separate different abstraction layers.

Reply:
BTW, it feels like calling `get_repo_path` for the second time for the same `dest` is somehow redundant. `get_repo_path` is called in `main` anyway. Is there a way to save the result after the first invocation and use it later?

Reply:
Yes, it's called [`lru_cache`](https://docs.python.org/3/library/functools.html#functools.lru_cache), but it's not available in stdlib of Python 2.7, so we cannot use it (unless we vendor [it's backport](https://pypi.org/p/backports.functools_lru_cache) into the project like `six` distribution).